src/components/reps/promise-handler.jsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+
+import { renderValue } from './value-renderer'
+
+export class PromiseRep extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      status: 'pending',
+      value: undefined,
+      runTime: 0,
+    }
+    this.state.promise = this.props.promise.then(
+      (val) => {
+        this.setState({ status: 'resolved', value: val })
+        return val
+      },
+      (val) => {
+        this.setState({ status: 'rejected', value: val })
+        return val
+      },
+    )
+    const runTimer = setInterval(() => {
+      if (this.state.status === 'pending') {
+        this.setState({ runTime: this.state.runTime + 1 })
+      } else {
+        clearInterval(runTimer)
+      }
+    }, 1000)
+  }
+
+  render() {
+    const { value, status, runTime } = this.state
+    let outputValue
+    let statusDisplay = status
+    if (status === 'resolved' || status === 'rejected') {
+      outputValue = <div className="promise-handler-value">{renderValue(value)}</div>
+    } else if (status === 'pending') {
+      statusDisplay = `pending (${runTime} sec)`
+    }
+    return (
+      <div>
+        <div className="promise-handler-state">
+          <div className="promise-handler-label">
+                Promise
+          </div>
+          <div className="promise-handler-status">
+            {statusDisplay}
+          </div>
+        </div>
+        {outputValue}
+      </div>
+    )
+  }
+}
+
+export default {
+  // based off of https://stackoverflow.com/questions/27746304/how-do-i-tell-if-an-object-is-a-promise
+  // It is unfortunate that this seems to be the main perspective for how to determine if
+  // something is Promise-ful. Ideally we would check only for native Promises,
+  // but at this point I don't know if others use Promise-based libraries.
+  shouldHandle: val => typeof val === 'object' && typeof val.then === 'function',
```

Comment:
as discussed-- let's only support built-ins
in that case, i think we should have:
`Promise.resolve().toString()` -> `"[object Promise]"`
(this is the case in Chrome, i haven't checked FF)